cmd/appliance/upgrade/prepare.go

```diff
@@ -164,6 +165,24 @@ func NewPrepareUpgradeCmd(f *factory.Factory) *cobra.Command {
 				}
 			}
 
+			if len(opts.logServerBundlePath) > 0 {
+				opts.logServerBundlePath = filesystem.AbsolutePath(opts.logServerBundlePath)
+				ok, err := util.FileExists(opts.logServerBundlePath)
+				if err != nil {
+					return err
+				}
+				if !ok {
+					errs = multierr.Append(errs, fmt.Errorf("Image bundle not found: %q", opts.logServerBundlePath))
+				}
+				if ok {
+					// validate bundle filename
+					regex := regexp.MustCompile(`logserver-\d{1,2}\.\d{1,2}(.+)?\.zip$`)
+					if !regex.MatchString(opts.logServerBundlePath) {
+						return fmt.Errorf("invalid bundle filename: %s. Expecting name matching pattern 'logserver-<MAJOR>.<MINOR>.zip'", opts.logServerBundlePath)
+					}
```

Comment:
I don't think the local filename should matter much, you can always pick the filename you want when uploading to the controller anyways.
Since it's a local file you can even inspect the file listing from the zip to check its the format we expect, or at least that its actually a zip file.